src/edge/simulator.ts

```diff
@@ -0,0 +1,191 @@
+            const version: string | null = Simulator.extractVersion(output);
+            if (version) {
+                const latestVersion: string | undefined = await Simulator.getLatestSimulatorVersion();
+                if (latestVersion && semver.gt(latestVersion, version)) {
+                    message = `${Constants.updateSimulatorMsg} (${version} to ${latestVersion})`;
+                }
+            } else {
+                message = Constants.updateSimulatorMsg;
+            }
+        } catch (error) {
+            message =  Constants.needSimulatorInstalledMsg;
+        }
+
+        if (message) {
+            await Simulator.installSimulatorWithPip(false, message, outputChannel);
+        }
+    }
+
+    private static iotedgehubdevVersionUrl = "https://pypi.org/pypi/iotedgehubdev/json";
+    private static learnMoreUrl = "https://github.com/Azure/iotedgehubdev#installing";
+
+    private static async validateSimulatorInstalled(outputChannel: vscode.OutputChannel = null): Promise<InstallReturn> {
+        if (await Simulator.simulatorInstalled()) {
+            return InstallReturn.Success;
+        } else {
+            return await Simulator.installSimulatorWithPip(true, Constants.needSimulatorInstalledMsg, outputChannel);
+        }
+    }
+
+    private static extractVersion(output: string): string | null {
+        if (!output) {
+            return null;
+        }
+        const pattern: RegExp = new RegExp(/version\s+([^\s]+)/g);
+        const matchRes = output.match(pattern);
+        if (matchRes !== null) {
+            return matchRes[0].replace(/version\s+/g, "");
+        }
+        return null;
+    }
+
+    private static async getLatestSimulatorVersion(): Promise<string | undefined> {
+        try {
+            const pipResponse = await request.get(Simulator.iotedgehubdevVersionUrl);
+            return JSON.parse(pipResponse).info.version;
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    private static async installSimulatorWithPip(force: boolean, message: string, outputChannel: vscode.OutputChannel = null): Promise<InstallReturn> {
+        let ret: InstallReturn = InstallReturn.Success;
+        if (await this.checkPipInstalled()) {
+            const install: vscode.MessageItem = { title: Constants.install };
+            const items: vscode.MessageItem[] = [ install ];
+            if (!force) {
+                items.push({ title: Constants.skipForNow });
+            }
+
+            let input: vscode.MessageItem | undefined;
+            if (force) {
+                input = await vscode.window.showWarningMessage(message, { modal: true }, ...items);
+            } else {
+                input = await vscode.window.showWarningMessage(message, ...items);
+            }
+
+            if (input === install) {
+                try {
+                    await Executor.executeCMD(outputChannel, "pip", {shell: true}, "install --upgrade iotedgehubdev");
```

Comment:
This may fail on *nix because of permission issue.

Reply:
pip install should not ask for sudo right? but run it needs sudo.

Reply:
Here is how vscode-python installs Python packages: https://github.com/Microsoft/vscode-python/blob/master/src/client/common/installer/moduleInstaller.ts#L36-L40

Reply:
If users have `python.pythonPath` set, the `iotedgehubdev start` command which stills runs in terminal, will pick up users `python.pythonPath` setting. Installing iotedgehubdev with output channel will have no effect.

Reply:
@adashen Whether `pip install` requires `sudo` really depends how users installs Python. For example, on macOS, if the user uses the built-in Python, `sudo` is required, but not for Homebrewed Python. This is the same for Linux.

Reply:
will check the platform, for windows install for user, else pop message and ask user to install manually.